feat(cocoapods): Add CocoaPods deprecation warnings to Podspec, README, and CHANGELOG (#16172)
firebase/firebase-ios-sdk: README.md

Author: Preview: <img width="1218" height="713" alt="Image" src="https://github.com/user-attachments/assets/99ead42b-a221-4428-8eb3-3d878d2dd037" />

Author: Could also do a purple "important" tag instead of the yellow "warning" one.

Source: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

Author: Going with [!IMPORTANT] because a yellow warning often implies an immediate break or danger. Since their current installations are safe today, this feels like the right tone—crucial for planning, but not a fire drill.

Author: <img width="1159" height="544" alt="Image" src="https://github.com/user-attachments/assets/e999f2e0-078a-4012-a353-939f33e1c4d3" />

  </a>
</p>

> [!WARNING]

> > **CocoaPods Deprecation:** The Firebase Apple SDK via CocoaPods is deprecated in favor of Swift

Author: ```suggestion
> **CocoaPods Deprecation:** Installing the Firebase Apple SDK using CocoaPods is deprecated in favor of Swift
```

  </a>
</p>

> > [!IMPORTANT]

Author: Open to feedback from other reviewers here. I'm not sure if warning comes across as more "scary" than warranted.

> > [!IMPORTANT]

Reviewer: I think warning is appropriate. We just had a small debacle over deprecating another product with not-scary-enough language and llm agents kept trying to use the deprecated product. Better safe than sorry.

  </a>
</p>

> - todo: deprecation announcement and link to migration guide

Author: ```suggestion
> [!WARNING]
> **CocoaPods Deprecation:** The Firebase Apple SDK via CocoaPods is deprecated in favor of Swift Package Manager. Firebase products remain fully supported, but new versions will no longer be published to CocoaPods after **October 2026**. See the [migration guide](https://firebase.google.com/docs/ios/cocoapods-deprecation) for more information.
```

>   </a>
> </p>
> 
> > [!WARNING]
> > **CocoaPods Deprecation:** The Firebase Apple SDK via CocoaPods is deprecated in favor of Swift Package Manager. Firebase products remain fully supported, but new versions will no longer be published to CocoaPods after **October 2026**. See the [migration guide](https://firebase.google.com/docs/ios/cocoapods-deprecation) for more information.
> 

Author: ```suggestion
> [!WARNING]
> **CocoaPods Deprecation:** The Firebase Apple SDK via CocoaPods is deprecated in favor of Swift
> Package Manager. Firebase products remain fully supported, but new versions will no longer be
> published to CocoaPods after **October 2026**. See the [migration guide](https://firebase.google.com/docs/ios/cocoapods-deprecation) for more information.

```
